gno.land/pkg/integration/testing_integration.go

```diff
@@ -284,6 +284,51 @@
 				genesis := ts.Value(envKeyGenesis).(*gnoland.GnoGenesisState)
 				genesis.Balances = append(genesis.Balances, balance)
 			},
+			// adduserfrom commands must be executed before starting the node; it errors out otherwise.
+			"adduserfrom": func(ts *testscript.TestScript, neg bool, args []string) {
+				if nodeIsRunning(nodes, getNodeSID(ts)) {
+					tsValidateError(ts, "adduserfrom", neg, errors.New("adduserfrom must be used before starting node"))
+					return
+				}
+
+				var account, index uint64
+				var err error
+
+				switch len(args) {
+				case 2:
```

Comment:
Maybe we should error out here for better UX?

Reply:
Looking at it again, `case 2:` means the user provided the following line in txtar:

`adduserfrom 'username 'menmonic'`.

So, the idea is to exit the switch and go straight to the creation of the account. Possibly it would be good to annotate each case with comments for easier readability.

Reply:
7e47fd9fdab5cd1ebcd723dca14d93910e2e246d